code/modules/reagents/Chemistry-Recipes.dm

```diff
@@ -602,26 +602,94 @@
 	G.adjust(0,0,0,vol)
 	..()
 
-/datum/chemical_reaction/plasmasolidification
+/datum/chemical_reaction/solidification
+	name = "Metal solidification"
+	id = "metalsolid"
+	result = null
+	required_reagents = list(SILICATE = 10, FROSTOIL = 10, IRON = 20)
+	result_amount = 1
+
+/datum/chemical_reaction/solidification/on_reaction(var/datum/reagents/holder, var/obj/item/stack/sheet/to_spawn)
+	var/location = get_turf(holder.my_atom)
+	new to_spawn(location, result_amount)
+
+/datum/chemical_reaction/solidification/plasma
 	name = "Solid Plasma"
 	id = "solidplasma"
 	result = null
-	required_reagents = list(IRON = 5, FROSTOIL = 5, PLASMA = 20)
+	required_reagents = list(SILICATE = 10, FROSTOIL = 10, PLASMA = 20)
 	result_amount = 1
 
-/datum/chemical_reaction/plasmasolidification/on_reaction(var/datum/reagents/holder, var/created_volume)
-	var/location = get_turf(holder.my_atom)
-	new /obj/item/stack/sheet/mineral/plasma(location)
+/datum/chemical_reaction/solidification/plasma/on_reaction(var/datum/reagents/holder, var/obj/item/stack/sheet/to_spawn)
+	to_spawn = /obj/item/stack/sheet/mineral/plasma
```

Comment:
you could have just put this path in the result variable that you're leaving null anyway like I suggested, there's no need for this extra var and override!

oh well, close enough I guess